code/modules/mob/living/living_say.dm

```diff
@@ -371,6 +371,11 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
 
 	var/list/listened = list()
 	for(var/atom/movable/listening_movable as anything in listening)
+		// SKYRAT EDIT DISABLE GHOST VIEWING
+		var/mob/listener = listening_movable
+		if(is_speaker_whispering && isobserver(listening_movable) && !listener?.client?.holder && HAS_TRAIT(src, TRAIT_NO_GHOST_MESSAGES))
```

Comment:
This should probably be moved up, where there's already an `is_speaker_whispering` check, as that would make this change far less intrusive and potentially hot.